### PR TITLE
Add Resend extension

### PR DESCRIPTION
**Pull request by xmok**
## Description

This is a Raycast extension for [Resend](https://resend.com/). With this extension you can Manage API Keys and Domains, and Send Emails using Resend.

## Screencast

https://github.com/raycast/extensions/assets/30526133/2db647ae-df52-49de-b57b-37e37808aeba

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by mil3na** [comment]
Hi! Thank you so much for your contribution to Raycast!
My name is Milena and I will be reviewing your extension today :)

I have a couple of concerns about running your extension for the first time. First, inserting the wrong API Key (or without enough permission) doesn't block the user from continuing the commands.  Maybe it would be better UX if the user would not be allowed to continue and ask for a correct key?
CleanShot 2023-06-27 at 13 33 59@2x

Also, I saw that you uploaded 3 flags for the domain locations. Are these the only locations provided by Resend? Or will they ever expand?

File: extensions/resend/package.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+      "name": "emails",
+      "title": "Emails",
+      "subtitle": "Resend",
+      "description": "View Logged Emails and Send A New Email",
+      "mode": "view"
+    }
+  ],
+  "dependencies": {
+    "@raycast/api": "^1.53.2",
```
Comment (mil3na) [inline extensions/resend/package.json:47]: Please update to the latest version of the API.

**Comment by mil3na** [comment]
Hey! Thanks for the updates!

I have a couple of more comments:

There's a typo on the send email screen:

CleanShot 2023-06-28 at 13 00 25@2x

After creating a key, I see the message below, however, on the screen that lists the APIs, I can both see and copy the created keys.

CleanShot 2023-06-28 at 12 59 14@2x

**Comment by xmok** [comment]
Great catch on the typo!

The screen with the API Keys only lists the API ID and Name; the actual token which starts with "re_" can not be retrieved again through the Resend API nor through Resend Dashboard as shown in below screenshots:

DIALOG ON CREATING API KEY IN DASHBOARD
resend-create-api-key-dashboard

VIEW API KEYS IN DASHBOARD
resend-api-keys-in-account
VIEW API KEYS IN EXTENSION
resend-api-keys-in-extension

**Comment by mil3na** [comment]
> Great catch on the typo!
> 
> The screen with the API Keys only lists the API ID and Name; the actual token which starts with "re_" can not be retrieved again through the Resend API nor through Resend Dashboard as shown in below screenshots:
> 
> DIALOG ON CREATING API KEY IN DASHBOARD resend-create-api-key-dashboard
> 
> VIEW API KEYS IN DASHBOARD resend-api-keys-in-account VIEW API KEYS IN EXTENSION resend-api-keys-in-extension

Ahh got it! Sorry for the misunderstanding! We are good to go, then.

**Comment by xmok** [comment]
Found and fixed another small typo 🙈. Thank you for your help!

**Review by mil3na — APPROVED** [review]
Awesome job! Typos are always hard to spot.

**Pull request merged**